R/round.r

```diff
@@ -76,25 +77,30 @@ floor_date <- function(x, unit = c("second","minute","hour","day", "week", "mont
 #' # "2009-09-01 CDT"
 #' ceiling_date(x, "year")
 #' # "2010-01-01 CST"
-ceiling_date <- function(x, unit = c("second","minute","hour","day", "week", "month", "year")) {
-	unit <- match.arg(unit) 
-  
+ceiling_date <- function(x, unit = c("second","minute","hour","day", "week", "month", "year", "quarter")) {
+	unit <- match.arg(unit)
+
 	if(!length(x)) return(x)
-  
+
   if (unit == "second") {
     second(x) <- ceiling(second(x))
     return(x)
   }
-  
+
+  if(unit != "quarter") {
 	y <- floor_date(x - dseconds(1), unit)
-	
+  } else {
+      y <- floor_date(x, unit)
+  }
+
 	switch(unit,
 		minute = minute(y) <- minute(y) + 1,
 		hour =   hour(y) <- hour(y) + 1,
 		day =    yday(y) <- yday(y) + 1,
 		week =   week(y) <- week(y) + 1,
 		month =  month(y) <- month(y) + 1,
-		year =   year(y) <- year(y) + 1
+		year =   year(y) <- year(y) + 1,
+        	quarter = month(y) <- month(y) + 3
```

Comment:
Oh wow, I just tried this out:
```
x <- ymd("2014-11-15")
month(x) <- month(x) + 3
x #=>  "2015-02-01 UTC"
```
That's pretty cool. I forget the OO terminology, but is it like `month` is an "active attribute" or something like that, such that if it is changed, the entire object is updated? I had assumed that it would return something like `month(x) = 14` after the update.